EasyCommands/BlockHandlers/LightBlockHandlers.cs

```diff
@@ -19,10 +19,39 @@
 
 namespace IngameScript {
     partial class Program {
+        public class SearchlightHandler : SubTypedBlockHandler<IMyFunctionalBlock> {
+            Dictionary<String, int> targetingGroupLookup = NewDictionary(
+                KeyValuePair("Default", 0),
+                KeyValuePair("Power", 1),
+                KeyValuePair("Weapons", 2),
+                KeyValuePair("Propulsion", 3));
+
+            public SearchlightHandler() : base(IsSubType("Searchlight")) {
+                AddPropertyHandler(Property.COLOR, TerminalPropertyHandler("Color", Color.Black));
+                AddPropertyHandler(Property.RADIUS, TerminalPropertyHandler("Radius", 10));
+                AddPropertyHandler(Property.RANGE, TerminalPropertyHandler("Range", 100));
+                AddPropertyHandler(Property.INTERVAL, TerminalPropertyHandler("Blink Interval", 0.1f));
+                AddPropertyHandler(Property.LEVEL, TerminalPropertyHandler("Blink Lenght", 0.1f));//Intentionally Typod
+                AddPropertyHandler(Property.OFFSET, TerminalPropertyHandler("Offset", 0.1f));
+                AddPropertyHandler(Property.VOLUME, TerminalPropertyHandler("Intensity", 1f));
+                AddPropertyHandler(Property.FALLOFF, TerminalPropertyHandler("Falloff", 0.5f));
+                AddPropertyHandler(Property.ROLL_INPUT, TerminalPropertyHandler("EnableIdleMovement", true));
+                AddPropertyHandler(Property.LOCKED, TerminalPropertyHandler("EnableTargetLocking", true));
```

Comment:
This property needs to be added to cockpits

Reply:
Added.  We should revisit and add "target locking" with the multi-property support.